### ISSUE TITLE
Can this program be run on an external hard drive?

### ISSUE THREAD

**Issue by JavierFFavieres** [gbouras13/hybracter#83]
My computer is running low on storage, so I was trying to run it on an external server comprised of hard drives. However, each time I get the error 'Snakemake failed'. 

In the computer itself it runs smoothly (with only a few genomes at a time), so I guess the program and its dependencies are properly installed, meaning the only difference is the location of the files.

Is there any way to circumvent this and run it on my server?

Thank you in advance,
Javier

**Comment by gbouras13** [comment]
Hi @JavierFFavieres 

This is very strange - I can't think of any reason it wouldn't work on external drives.

Can you give some more detail on the error e.g. what dependency specifically errors out?

George

**Comment by JavierFFavieres** [comment]
Yes, of course. Sorry for not elaborating earlier. These are the error messages:

Outputting passed long reads

WorkflowError:
Error recording metadata for finished job ([Errno 1] Operation not permitted: '/media/ARU/Javi_temp/Hybracter_test/.snakemake/metadata/tmpl_abuwj_.aHlicmFj'). Please ensure write permissions for the directory /media/ARU/Javi_temp/Hybracter_test/.snakemake
Removing output files of failed job filtlong since they might be corrupted:
hybracter_results/processing/qc/39CS_filt.fastq.gz, hybracter_results/versions/39CS/filtlong.version
Shutting down, this might take some time.
Exiting because a job execution failed. Look above for error message
Complete log: .snakemake/log/2024-07-22T101436.643248.snakemake.log
WorkflowError:
At least one job did not complete successfully.
[2024:07:22 10:30:14] ERROR: Snakemake failed

Javier

**Comment by gbouras13** [comment]
Hi @JavierFFavieres ,

The error stems from permissions on your harddrive, not Hybracter (I think!)

```
Error recording metadata for finished job ([Errno 1] Operation not permitted: '/media/ARU/Javi_temp/Hybracter_test/.snakemake/metadata/tmpl_abuwj_.aHlicmFj'). Please ensure write permissions for the directory /media/ARU/Javi_temp/Hybracter_test/.snake-make
```

You may need to allow write access on your disk - see e.g. https://askubuntu.com/questions/90339/how-do-i-set-read-write-permissions-my-hard-drives

George

**Comment by JavierFFavieres** [comment]
Thank you so much for clarifying. I will look into it.

Thanks a lot
Javier
